docs/insomnia/git-sync.md

```diff
@@ -25,22 +27,33 @@ When configuring a remote repository, you will be prompted for the following inf
 * **Author Name**: The Git author name to store with each commit.
 * **Author Email**: The Git author email to store with each commit.
 * **Username**: The Git author username to match with the authentication token.
-* **Authentication Token**: The token needed to authenticate with remote repository provider, such as GitHub or BitBucket. If you have two-factor authentication (2FA) enabled on your account, it is unlikely you will be able to use your username and password. Instead, generate a Personal Access Token or App Password (see list below for links to documentation for your Git system).
+* **Authentication Token**: The token needed to authenticate with remote repository provider, such as GitHub or BitBucket. If you have two-factor authentication (2FA) enabled on your account, it is unlikely you will be able to use your username and password. Instead, generate a personal access token or app password (see list below for links to documentation for your Git system).
+
+{:.alert .alert-primary}
+**Note**: You may still fail to set up Git Sync properly due to not enough or the wrong types of Git permissions.
 
-Find instructions on how to create a Personal Access Token or App Password on the following platforms:
+Find instructions on how to create a personal access token or app password on the following platforms:
 
 * [Github](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token)
+  * To push to and pull from a _public_ GitHub repo, scope at least `public_repo` when creating your personal access token.
 * [Gitlab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)
+  * To push to and pull from a _private_ GitLab repo, scope at least `api` when creating your personal access token.
 * [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/)
+  * To push and pull from a _private_ BitBucket repo, scope at least `Read` and `Write` in the `Repository` options when creating your app password.
 * [Bitbucket Server](https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html)
 
 Once complete, click **Done** and the repository settings will be persisted for future operations. The author details and token can be updated as needed.
 
-## Clone an Existing Repository
+{:.alert .alert-primary}
+**Note**: When linking to a GitHub repository, the default branch maybe appear on the Insomnia app as `master`. If you use another default, like `main`, click on the branch dropdown. Select **Branches**, and add your default branch name.
```

Comment:
I tested with a new repository that has a README, and it appears to recognize `main` properly. Ie. if your repository has a default branch, which is `main`, Insomnia will recognize that.

I then tested with a new repository with nothing in it - completely blank, no branches. This one is defaulted to `master` in Insomnia.

I think instead of putting this note in the docs it might need to be a bug/feature request 🤔